src/invidious/user/imports.cr

```diff
@@ -30,21 +30,75 @@ struct Invidious::User
       return subscriptions
     end
 
-    def parse_playlist_export_csv(user : User, raw_input : String)
-      # Split the input into head and body content
-      raw_head, raw_body = raw_input.strip('\n').split("\n\n", limit: 2, remove_empty: true)
+    def parse_playlist_export_csv(user : User, raw_input : String, filename : String)
+      LOGGER.trace("parse_playlist_export_csv: 01 raw_input '#{raw_input}'\n")
+
+      raw_head = "" # playlists.csv - info-line for a given playlist in the google export, that's copied above the actual playlist-infos and separated by an empty line from
+      raw_body = "" # the actual playlist content, ie. a list of videos
 
-      # Create the playlist from the head content
-      csv_head = CSV.new(raw_head.strip('\n'), headers: true)
-      csv_head.next
-      title = csv_head[4]
-      description = csv_head[5]
-      visibility = csv_head[6]
+      # remove superfluous \n instances in front of and after any non-\n text.. so .. a trim?
+      raw_input = raw_input.strip('\n')
 
-      if visibility.compare("Public", case_insensitive: true) == 0
-        privacy = PlaylistPrivacy::Public
+      # Split the input into head and body content
+      tmp = raw_input.split("\n\n", limit: 2, remove_empty: true)
+      if tmp.size > 1
+        raw_head = tmp[0]
+        raw_body = tmp[1]
       else
-        privacy = PlaylistPrivacy::Private
+        raw_body = tmp[0]
+      end
+
+      LOGGER.trace("parse_playlist_export_csv: 02 raw_head '#{raw_head}' -----\nraw_body '#{raw_body}'\n")
+
+      # TODO create an import-feature (elsewhere), which works for the original google export format, ie. the playlists/ subdirectory content as a ZIP file.. or a complete youtube music export file, but this goes way beyond the scope of a playlist import.
+
+      # defaults
+      title = "title not set"          # TODO i8n
+      description = "from #{filename}" # TODO i8n
+      visibility = "Private"
+      privacy = PlaylistPrivacy::Private
+
+      if (raw_head != "")
+        # ## XXX for documentation of current file format of playlists.csv (today it's 2024-01-07):
+        # 0           1                     2                                 3                    4                       5                      6                         7                                  8                         9                         10                   11
```

Comment:
If I'm understanding correctly this is basically supporting a "new format" that is produced by concatenating the specific playlist header from `playlists.csv` onto the csv file containing the videos for the corresponding playlist?